src/blocks/store/s3.clj

```diff
@@ -176,7 +176,9 @@
 (defrecord S3BlockStore
   [^AmazonS3 client
    ^String bucket
-   ^String prefix]
+   ^String prefix
+   ^boolean set-sse-flag?
+   ^clojure.lang.IFn mutate-put-metadata-fn]
```

Comment:
I'd like to simplify this name a bit - how about `alter-put-metadata`? You also don't need to type-hint these fields - the existing ones are hinted because they are used in various java interop calls.